contracts/governance/staking/StakedTokenBPT.sol

```diff
@@ -250,7 +243,41 @@ contract StakedTokenBPT is StakedToken {
     /**
      * @dev Get the current priceCoeff
      */
-    function _getPriceCoeff() internal view override returns (uint256) {
-        return priceCoefficient;
+    function _getPriceCoeff() internal view override returns (uint256 priceCoeff) {
+        priceCoeff = priceCoefficient;
+    }
+
+    /***************************************
+                BALANCER POOL GAUGE
+    ****************************************/
+
+    /**
+     * @dev Transfers staked tokens from sender to this staking token contract,
+     * deposits staked tokens in the Balancer Pool Gauge to earn rewards and
+     * finally call `_settleStake`.
+     */
+    function _transferAndStake(
```

Comment:
Should be there any validation to amount or delegatee ?  is there any limit that should be consider ?

Reply:
Good question. The old code didn't do any validation on the amount staked. The user can stake as much as they want. If they try and stake more than they have then the transfer from the msg sender into the staking contract will fail.

There is an existing check at the start of `StakedToken._settleStake` for a zero amount staked.

There is also no restriction on who the stkBPT tokens are delegated to. We could add basic checks like no ZERO addresses but what about other address like address(1)? Given we are already at the contract size limit, I'd prefer not to validate the `_delegatee` is a zero address.